README.md

```diff
@@ -1,2 +1,111 @@
+    + [20220322_compare_kmer_and_gene_count_dists.ipynb](notebooks/20220322_compare_kmer_and_gene_count_dists.ipynb)
+2. K-mer coverage for genes and operons. Do most genes and most operons have at least one hash (k-mer) in the FracMinHash sketch?
+    + [20220325_hashes_in_transcripts_and_operons.ipynb](notebooks/20220325_hashes_in_transcripts_and_operons.ipynb)
+3. Normalization techniques and correlation analysis. Can we recover co-expression patterns using correlation analysis of normalized k-mer abundances?
+    + DESeq2 median of ratios normalization: [20220328_correlations_mrnorm.ipynb](notebooks/20220328_correlations_mrnorm.ipynb)
+    + DESeq2 variance stabilized transformation: [20220330_correlations_vstnorm.ipynb](notebooks/20220330_correlations_vstnorm.ipynb)
+    + Seurat log normalization: [20220330_correlations_lognorm.ipynb](notebooks/20220330_correlations_lognorm.ipynb)
+    + Seurat centered log ratio normalization: [20220330_correlations_clrnorm.ipynb](notebooks/20220330_correlations_clrnorm.ipynb)
+    + Seurat counts per million normalization: [20220330_correlations_rcnorm.ipynb](notebooks/20220330_correlations_rcnorm.ipynb) 
+
+## Results and conclusions
+
+**Results**
+
+See notebooks for more details.
+
++ K-mer abundance distributions are weird
++ Many genes don't have a k-mer, but most operons do
++ Many spurious correlations remain after normalization, obscuring signal from intra-operon correlations. Median of ratios and centered log ratio normalization appear to have performed the best, however many spurious correlations remain after normalization. In situations where we won't have gene annotations available, this would make it difficult to determine which k-mers are actually part of the same operon and which are not. 
+
+**Conclusions**
+
+I don't think k-mer abundances can be used to recover co-expression patterns; too many spurious correlations remain after normalization.
+Brainstorming possible reasons why this might be:
+1. The RNAseq samples in the PA14 compendium may be too dissimilar, meaning that even at a k-mer size of 21 there are too few shared k-mers between many of samples to capture co-expression relationships. There is some evidence that this may be the case; the compendium was constructed using a k-mer size of 15 to seed the mappings. When the default k-mer size of 31 was used, the mapping rates were very low for most RNAseq samples. Dropping the k-mer size to 15 increased the mapping rate by 20%. This indicates that there are many SNPs in the RNAseq samples. Because k-mers have to exactly match to enable comparisons this may be artificially reducing gene-gene correlations.
+2. K-mer count distributions are weird, so normalizing them doesn't lead to intended outcomes. 
+3. K-mer abundances may not be reflective of gene abundances, since gene abundances are inferred from number of reads mapped and k-mer abundances are an exact count of that k-mer. 
```

Comment:
Just curious if think that filtering to k-mers that map to highly expressed genes/operons/regions (i'm not familiar with bacteria) would change the analysis results. Or if you just looked at the kmer abundances across samples for highly expressed genes, if they k-mers correlated with gene expression.

Reply:
I don't think filtering to highly expressed regions would change the analysis results, because I think the same problems would remain.

K-mers _should_ correlate with gene expression, so I think within one sample we could get good mapping between k-mer abundances and the genes those k-mers come from. But I think what might be happening is that these things are too distantly related to get enough matching k-mers across genes that are the same, which is obscuring the results.